android/java/org/chromium/chrome/browser/rate/BraveAskPlayStoreRatingDialog.java

```diff
@@ -33,9 +34,27 @@
 public class BraveAskPlayStoreRatingDialog extends BottomSheetDialogFragment {
     final public static String TAG_FRAGMENT = "brave_ask_play_store_rating_dialog_tag";
     private static final String TAG = "AskPlayStoreRating";
+    private ReviewManager mReviewManager;
+    private ReviewInfo mReviewInfo;
+    private boolean mIsFromSettings;
+    private Context mContext;
 
-    public static BraveAskPlayStoreRatingDialog newInstance() {
-        return new BraveAskPlayStoreRatingDialog();
+    public static BraveAskPlayStoreRatingDialog newInstance(boolean isFromSettings) {
+        Bundle bundle = new Bundle();
+        bundle.putBoolean(RateUtils.FROM_SETTINGS, isFromSettings);
+        BraveAskPlayStoreRatingDialog fragment = new BraveAskPlayStoreRatingDialog();
+        fragment.setArguments(bundle);
+
+        return fragment;
+    }
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        mContext = context;
+        if (getArguments() != null) {
+            mIsFromSettings = getArguments().getBoolean(RateUtils.FROM_SETTINGS);
```

Comment:
Do we want to have this code here ? i think we decided to redirect to OS settings page. will be fixing it in a follow up ?

Reply:
Nevermind it's regarding the dialog.